AdblockPlusSafari/AppDelegate.m

```diff
@@ -153,11 +159,11 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
         for (NSString *filterListName in filterLists) {
           NSDictionary<NSString *, id> *filterList = filterLists[filterListName];
           // The date of the last known successful update
-          NSDate *lastUpdate = filterList[@"lastUpdate"];
+          NSDate *lastUpdate = filterList.lastUpdate;
 
-          NSDate *currentLastUpdate = self.adblockPlus.filterLists[filterListName][@"lastUpdate"];
+          NSDate *currentLastUpdate = self.adblockPlus.filterLists[filterListName].lastUpdate;
 
-          updated = updated || (!lastUpdate && currentLastUpdate) || (currentLastUpdate && [currentLastUpdate compare:lastUpdate] == NSOrderedDescending);
+          updated = updated || (!lastUpdate && !!currentLastUpdate) || (!!currentLastUpdate && [currentLastUpdate compare:lastUpdate] == NSOrderedDescending);
```

Comment:
Why add the `!!` here? I don't think it changes the behaviour of this expression.